src/sagemaker_training/smdataparallel.py

```diff
@@ -185,9 +187,12 @@ def _get_mpirun_command(
         mpirun_command.extend(additional_options)
 
         instance_type = self._get_instance_type()
-        # Use EFA's RDMA functionality for one-sided and two-sided transfer
+        # EFA settings
         if instance_type in ["ml.p3dn.24xlarge", "ml.p4d.24xlarge"]:
```

Comment:
EFA supports more GPU instances
find the full list here:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html

this will prevent other PT DDP instances to use EFA, like g5.

Although SMDDP only supports p3dn and p4d that use EFA. The PT DDP launcher is also utilizing the SMDDPRunner, we should support full EFA enabled instances here as well. Same criteria as mpi.py

Reply:
updated the list and made it common for all mpi based runs.